Articles/ShortArticles/DiscontinuingSoftwareProjects.md

```diff
@@ -0,0 +1,144 @@
+   Include key design decisions, architecture notes, algorithms, and implementation details that may be valuable to others.
+   If there is known to be large swaths of documentation that are sitting around in restricted places (emails, slide decks, pdf files) that are not already available on line and/or *with* the source code, consider getting this information *out there* with the source code as well maybe by adding the relevant artifacts to the repository holding the source code.
+
+1. Establish an enduring on-line presence
+
+   If the project doesn't already have one, create a web site.
+   Even a single page site that describes the software at a high level would be fine for this purpose.
+   There are various free website hosting options.
+   A good one is a [GitHub pages](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#) site because it offers a place to host both a web site about the project and the project's actual source code as well as other project artifacts.
+
+1. Archive the software
+
+   Consider archiving the software with a on line archival service.
+   There are archival services available such as [DOE Code](https://www.code.gov), [netlib.org](https://netlib.org), [Software Heritage](https://www.softwareheritage.org), [Internet Archive](https://archive.org) and [Archive Team](https://wiki.archiveteam.org/index.php/Main_Page) all of which handle archival of software among other things.
+   Another option to consider is archiving the software with a [Digital Object Identifier](https://bssw.io/items/persistent-identifiers-for-software-in-scientific-computing) (DOI) agency.
+   There are many DOI agencies often specializing in certain kinds of digital assets.
+   Some handle source code.
+   Consider [Crossref](https://www.crossref.org), [Zenodo](https://zenodo.org) and [Open Science Framework](https://osf.io).
+
+1. Present or Publish
+
+   Try to get at least one piece of literature (conference presentation or proceedings publication) anywhere it may be a good fit where you can write in more detail about the computer science known-how.
+   Often case-studies and lessons learned are good candidate topical areas for a piece of literature focusing on the computer science aspects of the project.
+   Published research papers, technical reports, or white papers that discuss the project's methodologies, challenges, and outcomes are a great way to capture the computer science knowledge of a project.
+
+1. Refactor critical dependencies
+
+   Instead of the *whole* code base, it may be appropriate to carve out *pieces* of the code for the actions we're discussing here.
+   Assess if any parts of the code base or architecture can be reused in other projects.
+   Salvaging valuable components can save time and effort in future development.
+   Depending on how the software is architected, the work to refactor any such pieces of the code may be doable before funding runs out.
```

Comment:
Or, it may be enough to just document what parts of the software might be good candidates to be factored out of the code and reused.  It is usually better to not refactor a piece of software for resuse until there is actually a need to do so.